src/java.base/macosx/classes/sun/nio/fs/BsdFileSystem.java

```diff
@@ -71,13 +76,80 @@ public Set<String> supportedFileAttributeViews() {
         return SupportedFileFileAttributeViewsHolder.supportedFileAttributeViews;
     }
 
+    /**
+     * Clones the file whose path name is {@code src} to that whose path
+     * name is {@code dst} using the {@code clonefile} system call.
+     *
+     * @param src the path of the source file
+     * @param dst the path of the destination file (clone)
+     * @param followLinks whether to follow links
+     *
+     * @return 0 on success, or IOStatus.UNSUPPORTED_CASE if the call
+     *         does not work with the given parameters
+     */
+    private int clone(UnixPath src, UnixPath dst, boolean followLinks)
+        throws IOException
+    {
+        int flags = followLinks ? 0 : CLONE_NOFOLLOW;
+        try {
+            BsdNativeDispatcher.clonefile(src, dst, flags);
+        } catch (UnixException x) {
+            // clone failed so roll back
+            try {
+                unlink(dst);
+            } catch (UnixException ignore) { }
```

Comment:
My previous comment about the missing unlink was a comment on BsdFileSystem.copyFile where it will leave dst in place when the chown fails. The unlink you've added to clone is harmless but the right place is copyFile so that it attempts to cleanup/unlink when an exception is thrown.

Reply:
Fixed in 96db8562fad4396500f8386c5f26c5fa98b7c373.